### PR TITLE
shared/install: fix error condition for symlink creation

### PR DESCRIPTION
**Pull request by eworm-de**

### CODE REVIEW

File: src/shared/install.c [hunk 1]
```diff
@@ -1492,7 +1492,7 @@ static int install_info_symlink_alias(
                 rp = skip_root(paths, i->path);
 
                 q = create_symlink(rp ?: i->path, alias_path, force, changes, n_changes);
-                if (r == 0)
+                if (q < 0)
                         r = q;
```
Comment (poettering) [inline src/shared/install.c:1496]: the old code was right here. The logic is to return the first error that happens, but continue operation. Thus, only when r is 0 we should override it.

(note that OOM is considered fatal here, only the symlink creation is something where we proceed after the first failure, but still return the first failure)

Comment (eworm-de) [inline src/shared/install.c:1496]: But the condition is always true, no? There is no code path where `r != 0`.

And `create_symlink()` returns 1 if a symlink was created or already exists. So `install_info_symlink_alias()` returns failure when it should not.

Comment (poettering) [inline src/shared/install.c:1496]: yeah, it probably should indeed say: `if (r == 0 || (r > 0 && q < 0)) r = q;`

i.e. as long as r is 0 the result should always be assigned. If r is > 0 it should only be assigned if failure.

This means: the whole funciton returns 0 if nothing happened, 1 if something happened and the first error if an error happened.

Does that make sense?

**Comment by eworm-de** [comment]
That does not change anything as `r == 0` is still always true.

File: src/shared/install.c (hunk 1 above)
Comment (eworm-de) [inline src/shared/install.c:1496]: That does not change anything as `r == 0` is still always true.

Comment (poettering) [inline src/shared/install.c:1496]: it's going to be 0 initialially. But this is a loop, and as soon as the if check hits once, then r is overriden with q, which might not be 0 anymore... So no, this is not always true...

**Pull request closed without merging**